README.md

```diff
@@ -1,15 +1,106 @@
-- dependabot setup with automatic merging thanks to ["merge dependabot" GitHub action](https://github.com/fastify/github-action-merge-dependabot)
-- notifications about commits waiting to be released thanks to ["notify release" GitHub action](https://github.com/nearform/github-action-notify-release)
-- PRs' linked issues check with ["check linked issues" GitHub action](https://github.com/nearform/github-action-check-linked-issues)
-- Continuous Integration GitHub workflow
+## Installation
 
-## When you have already a repo
+```bash
+npm i fastify-slow-down
+```
 
-If you already created a repo and you want to add some of the features above to it, you can have a look at [NearForm MRM Preset](https://github.com/nearform/mrm-preset-nearform).
+## Usage
+Register the plugin.
+This plugin will add an `onRequest` hook to slow down replies if a client (based on their IP address) has made too many multiple requests in the given `timeWindow`.
+```js
+import Fastify from 'fastify'
+import slowDownPlugin from 'fastify-slow-down'
+
+const fastify = Fastify()
+
+// register the plugin
+fastify.register(slowDownPlugin)
+
+// create a route
+fastify.get('/', async (req, reply) => 'Hello fastify-slow-down!')
+
+// start server
+await fastify.listen({ port: 3000 })
+```
+
+The response will have some additional headers:
+
+| Header | Description |
+|--------|-------------|
+|`x-slow-down-limit`     | how many requests in total the client can make until the server starts to slow down the response
+|`x-slow-down-remaining` | how many requests remain to the client in the `timeWindow`
+|`x-slow-down-delay` | how much delay (in milliseconds) has been applied to this request
+
+
+## Configuration
+- **delay**: Amount of delay imposed on request. it can be expressed in miliseconds or as string ([`ms`](https://github.com/zeit/ms) format). Set to `0` to disable delaying.
+
+- **timeWindow**: The duration of the time window. It can be expressed in milliseconds or as a string ([`ms`](https://github.com/zeit/ms) format). Set to `0` to disable delaying.
+
+- **delayAfter**:  max number of connections during `timeWindow` before starting to delay responses with time specifed `delay`. Set to `0` to disable delaying.
+
+- **keyGenerator**: Function used to generate keys for saved records in memory.
+
+
+## Example with configuration
+Giving the options it can be configurable for example: 
+```js
+{ 
+    delay: '10 seconds',
+     delayAfter: 10,
+     timeWindow: '10 minutes'
+}
+```
+
+The `delay` option will be applyied to the server when it is hitted for more then `delayAfter` option times in the specific `timeWindow` option
+
+
+***
+In 10 minutes the result of hitting the API will be: 
+
+1st request - no delay
+
+2nd request - no delay
+
+3rd request - no delay
+
+```...```
+
+10 request - no delay
+
+11 request - 10 seconds  delay
+
+12 request - 20 seconds  delay
+
+13 request - 30 seconds  delay
+
+```...```
+
+20 request - 100 seconds  delay
+
+```...```
+
+
+***
+
```

Comment:
let's format this in a slightly better way, for example a unordered list, and explain that after 10 minutes, no delay at all is applied, meaning that the timeWindow is a fixed window, not a sliding window